src/main/java/cubicchunks/server/ServerCubeCache.java

```diff
@@ -67,59 +65,55 @@
  * (there may be some entities that are not in any Cube yet).
  * * dropChunk method is not supported. Columns are unloaded automatically when the last cube is unloaded
  */
-public class ServerCubeCache extends ChunkProviderServer implements ICubeCache {
+public class ServerCubeCache extends ChunkProviderServer implements ICubeCache, IProviderExtras{
 
 	private static final Logger log = CubicChunks.LOGGER;
 
 	public static final int SPAWN_LOAD_RADIUS = 12; // highest render distance is 32
 
 	private ICubicWorldServer worldServer;
 	private CubeIO cubeIO;
-	private HashMap<Long, Column> loadedColumns;
 	private Queue<CubeCoords> cubesToUnload;
 	private Queue<ChunkPos> columnsToUnload;
 
-	public ServerCubeCache(ICubicWorldServer worldServer) {
-		//TODO: Replace add ChunkGenerator argument and use chunk generator object for generating terrain?
-		//ChunkGenerator has to exist for mob spawning to work
-		super((WorldServer) worldServer, worldServer.getSaveHandler().getChunkLoader(worldServer.getProvider()),
-				new ChunkProviderOverworld((World) worldServer, worldServer.getSeed(), false, null));
+	private Map<CubeCoords, Cube> cubemap = new HashMap<>();
+
+	private ICubeGenerator   cubeGen;
+
+	public ServerCubeCache(ICubicWorldServer worldServer, ICubeGenerator cubeGen) {
+		super((WorldServer) worldServer,
+				worldServer.getSaveHandler().getChunkLoader(worldServer.getProvider()), // forge uses this in
+				null); // safe to null out IChunkGenerator
```

Comment:
It is safe because I did not crash when I tested it... (and maybe overriding the eye of ender looking for stronghold method might have something to do with it)
I will add a bit more to the comment.